server/block/model/trapdoor.go

```diff
@@ -19,14 +19,19 @@ type Trapdoor struct {
 // top part of the block.
 func (t Trapdoor) BBox(cube.Pos, *world.World) []cube.BBox {
 	if t.Open {
-		return []cube.BBox{full.ExtendTowards(t.Facing.Face(), -0.8125)}
+		return []cube.BBox{full.ExtendTowards(t.Facing.Face().Opposite(), -0.8125)}
```

Comment:
The original calculation seems to be correct. You can test by shooting a bow at opened trapdoors.